modules/local/validatemodel/resources/usr/bin/validate_model.R

```diff
@@ -0,0 +1,433 @@
+        Will also look for unwanted characters within columns and colnames.
+        Finally, will evaluate all model designs based on the formula and contrasts to find whether the models are full ranked or not.
+        "
+    )
+opt <- parse_args(opt_parser)
+
+## Validate input arguments
+### Required arguments
+if (is.null(opt$yml) || is.null(opt$samplesheet) ) {
+    stop("'--yml' and '--samplesheet' arguments are required.", call. = FALSE)
+}
+
+### Collect parameters (easier for dev and testing)
+path_yml         <- opt$yml             # "/workspace/differentialabundance/modules/local/validatemodel/tests/contrasts.yml" #
+path_samplesheet <- opt$samplesheet     # "/workspace/differentialabundance/results/SRP254919.samplesheet_witherrors.csv" #
+sample_column    <- opt$sample_id_col   # "sample" #
+
+# LOAD FILES --------------------------------------------------------
+## Load models.yml file
+tryCatch({
+    models <- read_yaml(path_yml)
+    cat("Loaded YML file successfully.\n")
+}, error = function(e) {
+    stop("Error loading YML file: ", e$message)
+})
+
+## Load samplesheet CSV file
+tryCatch({
+    samplesheet <- readr::read_csv(path_samplesheet, show_col_types = FALSE)
+    if ( !sample_column %in% colnames(samplesheet) ) {
+        stop(paste0("Column ", sample_column, " not present in colnames(samplesheet)") )
+    }
+    cat("Loaded samplesheet successfully.\n")
+}, error = function(e) {
+    stop("Error loading samplesheet CSV: ", e$message)
+})
+
+# PARSE FACTORS/LEVELS FROM YML FILE --------------------------------
+## Collect all variables and factors from the yml file
+## * var_list:        collects all factors levels for each variable.
+## * contrast_list:   collects info for each contrast
+
+process_models <- function(models) {
+    # Initialize lists to store the output
+    contrasts_list <- list()
+    var <- list()
+
+    # Ensure models$models is not null or empty
+    if (is.null(models$models) || length(models$models) == 0) {
+        stop("models$models is null or empty. Please provide valid input.")
+    }
+
+    # Temporary storage for gathering contrasts per variable
+    temp_var <- list()
+    blocking_vars <- c()
+
+    # Iterate through models (formula)
+    for (FORMULA in models$models) {
+
+        # Ensure FORMULA$contrasts is not null or empty
+        if (is.null(FORMULA$contrasts) || length(FORMULA$contrasts) == 0) {
+            stop("FORMULA$contrasts is null or empty. Unable to proceed with this model.")
+        }
+
+        # Iterate through contrasts for each model (formula)
+        for (CONTRAST in FORMULA$contrasts) {
+
+            # Validate CONTRAST$id
+            if (is.null(CONTRAST$id)) {
+                stop("Missing CONTRAST$id detected. Unable to proceed.")
+            }
+
+            if (CONTRAST$id %in% names(contrasts_list)) {
+                stop(paste("Duplicate CONTRAST$id detected:", CONTRAST$id, ". Unable to proceed."))
+            }
```

Comment:
Some of these things could already be guaranteed by the json schema for the contrasts definition (#405). Maybe not necessary to test it again here (but also doesn't harm)

Reply:
@grst Thanks for the input! If we don't find a reason to remove them, I'd like to keep these checking points within the script, just in case someone uses it outside the pipeline, or even for nf-tests and dev